### PR TITLE
Cherry pick transformation framework

### PR DESCRIPTION
**Pull request by seanzhougoogle**

### CODE REVIEW

File: delta-api/pom.xml [hunk 1]
```diff
@@ -36,6 +36,12 @@
       <artifactId>cdap-api</artifactId>
       <version>${cdap.version}</version>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-api-common</artifactId>
+      <version>6.6.0-SNAPSHOT</version>
```
Comment (sagarkapare) [inline delta-api/pom.xml]: `${cdap.version}`?

File: delta-api/pom.xml [hunk 2]
```diff
@@ -36,6 +36,12 @@
...
+      <scope>compile</scope>
```
Comment (sagarkapare) [inline delta-api/pom.xml]: I think `compile` is default scope, you dont need to specify explicitly.

**Review by sagarkapare — APPROVED** [review]
LGTM on cherry-pick assuming no conflicts.

File: delta-api/pom.xml (hunk 1 above)
Comment (seanzhougoogle) [reply delta-api/pom.xml]: done :)

File: delta-api/pom.xml (hunk 2 above)
Comment (seanzhougoogle) [reply delta-api/pom.xml]: done :)

**Pull request merged**